src/globus_cli/parsing/param_classes.py

```diff
@@ -64,14 +100,13 @@ def one_use_option(*args, **kwargs):
     if kwargs.get("is_flag"):
         kwargs["is_flag"] = False  # mutually exclusive with count
         kwargs["count"] = True
+        if "type_annotation" not in kwargs:
+            kwargs["type_annotation"] = bool
```

Comment:
Why?

Reply:
I think this deserves a comment. `one_use_option` is a bit weird (and maybe we should remove it in the future).
It takes any option and converts it to a form which allows us to count and detect multiple usages.

I would describe it as doing the following two transformations:
```python
# --foo is a bool flag
@one_use_option("--foo", is_flag=True)

# rewrites to --foo as a count which converts back to bool
def callback(ctx, param, value):
    if value > 1:
        raise UsageError("bad")
    return bool(value)

@click.option("--foo", count=True, callback=callback)
```
and
```python
# --foo is a non-flag option
@one_use_option("--foo")

# rewrites to --foo as a multiple option which unpacks its only value
def callback(ctx, param, value):
    if len(value) > 1:
        raise UsageError("bad")
    if value:
        return value[0]
    return None

@click.option("--foo", multiple=True, callback=callback)
```

This means that the type checker won't see `is_flag=True` on the created option, but `count=True` instead. Under normal logic, as for `click.option("-v", "--verbose", count=True)`, it will determine that the type is `int`. But that "hidden" callback in `one_use_option` makes it `bool`.

Reply:
Upon review, I decided to pull this out and instead pass `type_annotation=bool` in the 3 relevant usage sites.
I considered making the OneUseOption class responsible for this and various other "clever" solutions.

None of them stacked up, IMO, well against accepting that if you want to use `one_use_option("--foo", is_flag=True)`, then you must additionally pass `type_annotation=bool`. There's less to explain this way, which I see as the primary advantage.